aria_data_loaders/aria_data_utils/aria_streamer.py

```diff
@@ -0,0 +1,842 @@
+import os
+import time
+from typing import Any, Dict, List, Optional
+
+import cv2
+import numpy as np
+import sophus as sp
+from fairotag.scene import Scene
+from matplotlib import pyplot as plt
+from projectaria_tools.core import calibration, data_provider, mps
+from scipy.spatial.transform import Rotation as R
+from spot_rl.envs.skill_manager import SpotSkillManager
+
+### - $$$ SPOT=start $$$
+from spot_wrapper.spot import Spot, SpotCamIds, image_response_to_cv2
+
+from aria_data_loaders.aria_data_utils.utils.april_tag_pose_estimator import (
+    AprilTagPoseEstimator,
+)
+
+### - $$$ SPOT=end $$$
+
+"""
+dict = {
+    "file" : "Start_2m.vrs",
+    "2m" : 245,
+    "3m" : 334,
+    "4m" : 463,
+    "5m" : 574,
+}
+
+dict = {
+    "file" : "Start_1halfm.vrs",
+    "2m" : 238,
+    "3m" : 348,
+    "4m" : 458,
+    "5m" : 598,
+}
+
+dict = {
+    "file" : "Start_2m.vrs",
+    "2m" : 237,
+    "3m" : 337,
+    "4m" : 457,
+    "5m" : 581,
+}
+"""
+name = "Start_2m"
+vrsfile = f"/home/kavitsha/fair/aria_data/{name}/{name}.vrs"
+mpspath = f"/home/kavitsha/fair/aria_data/{name}/"
+DOCK_ID = int(os.environ.get("SPOT_DOCK_ID", 520))
+AXES_SCALE = 0.9
+STREAM1_NAME = "camera-rgb"
+STREAM2_NAME = "camera-slam-left"
+STREAM3_NAME = "camera-slam-right"
+
+############## Simple Helper Methods to keep code clean ##############
+
+
+def label_img(
+    img: np.ndarray,
+    text: str,
+    org: tuple,
+    font_face: int = cv2.FONT_HERSHEY_SIMPLEX,
+    font_scale: float = 0.8,
+    color: tuple = (0, 0, 255),
+    thickness: int = 2,
+    line_type: int = cv2.LINE_AA,
+):
+    cv2.putText(
+        img,
+        text,
+        org,
+        font_face,
+        font_scale,
+        color,
+        thickness,
+        line_type,
+    )
+
+
+def decorate_img_with_text(img, frame: str, position):
+    label_img(img, "Detected QR Marker", (50, 50), color=(0, 0, 255))
+    label_img(img, f"Frame = {frame}", (50, 75), color=(0, 0, 255))
+    label_img(img, f"X : {position[0]}", (50, 100), color=(0, 0, 255))
+    label_img(img, f"Y : {position[1]}", (50, 125), color=(0, 250, 0))
+    label_img(img, f"Z : {position[2]}", (50, 150), color=(250, 0, 0))
+
+    return img
+
+
+# DEBUG FUNCTION. REMOVE LATER
+def take_snapshot(spot: Spot):
+    resp_head = spot.experiment(is_hand=False)
+    cv2_image_head_r = image_response_to_cv2(resp_head, reorient=True)
+
+    resp_hand = spot.experiment(is_hand=True)
+    cv2_image_hand = image_response_to_cv2(resp_hand, reorient=False)
+    cv2.imwrite("test_head_right_rgb1.jpg", cv2_image_head_r)
+    cv2.imwrite("test_hand_rgb1.jpg", cv2_image_hand)
+
+
+######################################################################
+
+
+class AriaReader:
```

Comment:
add docstring? Also consider changing name later to something like AriaQRReader, if its appropriate?

Reply:
Adding docstrings, but not the name as class is more generic than just reading QR from images?

Reply:
added